package NBlenderUserFactors.NProto;

message TFactor {
    optional float Value = 1;
    enum EFactorType {
        FT_UNKNOWN_TYPE = 0;
        FT_SHOW_TYPE = 1;
        FT_CLICK_TYPE = 2;
        FT_CLICK_WIN_TYPE = 3;
        FT_CLICK_LOSS_TYPE = 4;
        FT_NOCLICK_WIN_TYPE = 5;
        FT_CROSS_LOSS_TYPE = 6;
        FT_INVERSION_LOSS_TYPE = 7;
        FT_RERE_LOSS_TYPE = 8;
        FT_SHORT_CLICK_PENALTY_TYPE = 9;
        FT_WIN_TYPE = 10;
        FT_LOSS_TYPE = 11;
    }
    optional EFactorType Type = 2;
}

message TIntentFactors {
    repeated TFactor Factors = 1;
    optional string Intent = 2;
    optional uint32 Version = 3 [default = 0];
    optional bool Disabled = 4;
}

message TEmbeds {
    optional bytes Embed = 1;
    optional string FactorId = 2;
    optional fixed32 LastUpdateTimestamp = 4;
}

message TUserFactors {
    repeated TIntentFactors IntentFactors = 1;
    repeated TEmbeds Embeds = 2;
}
